crates/krun/src/guest/bin/krun-guest.rs

```diff
@@ -42,7 +42,12 @@ fn main() -> Result<()> {
 
     setup_fex()?;
 
-    configure_network()?;
+    let rt = tokio::runtime::Runtime::new().unwrap();
+    rt.block_on(async {
```

Comment:
If I understand the docs correctly, this will NOT have the desired effect of ensuring the safety of the `env::set_var` calls.

> This runs the given future on the current thread, blocking until it is complete, and yielding its resolved result. Any tasks or timers which the future spawns internally will be executed on the runtime.

> When the multi thread scheduler is used this will allow futures to run within the io driver and timer context of the overall runtime.
>
> Any spawned tasks will continue running after block_on returns.

https://docs.rs/tokio/latest/tokio/runtime/struct.Runtime.html#method.block_on

Reply:
I think you're right. Another approach could be, since I don't think any of the other tasks `krun-guest` puts in motion depends on the network, simply move the network configuration to `krun-server` which already uses tokio and doesn't use `env::set_var`.